blockonomics-woocommerce.php

```diff
@@ -155,30 +142,16 @@ function add_page()
             'Blockonomics', 'Blockonomics', 'manage_options',
             'blockonomics_options', 'show_options'
         );
-
-        if (get_option('blockonomics_api_updated') == 'true' && isset($_GET['settings-updated']) ? $_GET['settings-updated'] : '' == 'true')
-        {
-            $message = __('API Key updated! Please click on Test Setup to verify Installation. ', 'blockonomics-bitcoin-payments');
-            display_admin_message($message, 'updated');
-        }
-
-        if (isset($_POST['runTest']))
+        
+        if (get_option('blockonomics_redirect') == "yes")
         {
-            $setup_errors = $blockonomics->testSetup();
-
-            if($setup_errors)
-            {
-                display_admin_message($setup_errors, 'error');
-            }
-            else
-            {
-                $message = __('Congrats ! Setup is all done', 'blockonomics-bitcoin-payments');
-                display_admin_message($message, 'updated');
-
-                $message = $blockonomics->make_withdraw();
-                if ($message) {
-                    display_admin_message($message[0], $message[1]);
-                }
+            update_option('blockonomics_redirect', 'no');
+            if(isset($_GET['tab']) && $_GET['tab'] == "currencies"){
+                wp_redirect( home_url('/wp-admin/options-general.php?page=blockonomics_options&tab=settings') ); 
```

Comment:
The admin settings are functioning well and the UI is complete. One last improvement would be to use `admin_url( 'options-general.php?page=blockonomics_options&tab=settings
' );` instead of `home_url()` to form the redirection URL.

Reply:
Makes sense. Thanks for the tip. I've committed and pushed the changes.